No warning for unused import in `__init__.py`

### Bug description


In a directory `tmp` I put the following code as `__init__.py`:


```python
"""Test module"""
import sys
```

Pylint 3.3.3 reports no problems with this code:

```sh
$ pylint __init__.py 

--------------------------------------------------------------------
Your code has been rated at 10.00/10 (previous run: 0.00/10, +10.00)
```

This is odd, as I have an unused import.

If I copy the same code to `test.py` and rerun pylint, I get the expected error:

```sh
$ pylint test.py
************* Module tmp.test
test.py:2:0: W0611: Unused import sys (unused-import)

-----------------------------------
Your code has been rated at 0.00/10
```

### Configuration

```ini

```

### Command used

```shell
pylint test.py
```

### Pylint output

```sh
--------------------------------------------------------------------
Your code has been rated at 10.00/10 (previous run: 0.00/10, +10.00)
```

### Expected behavior

```sh
__init___.py:2:0: W0611: Unused import sys (unused-import)

-----------------------------------
Your code has been rated at 0.00/10
```

### Pylint version

```shell
pylint 3.3.3
astroid 3.3.8
Python 3.12.3 (main, Nov  6 2024, 18:32:19) [GCC 13.2.0]
```

### OS / Environment

Ubuntu 24.04

### Additional dependencies

```python

```

--- a/doc/data/messages/u/unused-import/details.rst
+++ b/doc/data/messages/u/unused-import/details.rst
@@ -1,0 +1,1 @@
+By default, this check is skipped for ``__init__.py`` files, as they often contain imports from submodules for the convenience of end users. While these imports are not used within ``__init__.py``, they serve the purpose of providing intuitive import paths for the module's important classes and constants.

--- a/doc/data/messages/u/unused-import/related.rst
+++ b/doc/data/messages/u/unused-import/related.rst
@@ -1,0 +1,1 @@
+- `The --init-import configuration option <https://pylint.readthedocs.io/en/latest/user_guide/configuration/all-options.html#init-import>`_

--- a/doc/data/messages/u/unused-import/related.rst
+++ b/doc/data/messages/u/unused-import/related.rst
@@ -1,1 +1,1 @@
-- `The --init-import configuration option <https://pylint.readthedocs.io/en/latest/user_guide/configuration/all-options.html#init-import>`_
+- :ref:`--init-import <variables-options>`

